src/main/groovy/com/github/jrubygradle/jar/JRubyJarPlugin.groovy

```diff
@@ -23,45 +24,25 @@ class JRubyJarPlugin implements Plugin<Project> {
         project.apply plugin : 'java-base'
         project.configurations.maybeCreate('jrubyEmbeds')
         project.configurations.maybeCreate('jrubyJar')
+        project.tasks.create( 'jrubyJar',JRubyJar)
 
         updateTestTask(project)
-        addDependentTasks(project)
         addJrubyExtensionToJar(project)
         addAfterEvaluateHooks(project)
     }
 
     @PackageScope
-    void addDependentTasks(Project project) {
-        try {
-            Task t = project.tasks.getByName('jar')
-            if( t instanceof Jar) {
-                t.dependsOn 'jrubyPrepare'
-            }
-        } catch(UnknownTaskException) {
-            project.tasks.whenTaskAdded { Task t ->
-                if (t.name == 'jar' && t instanceof Jar) {
-                    project.task('jrubyJar', type: Jar) {
-                        dependsOn 'jrubyPrepare'
-                        destinationDir = t.destinationDir
-                        baseName = t.baseName
-                        extension = t.extension
-                        version = t.version
-                        appendix = 'all'
-                        from project.file("${project.buildDir}/dirinfo")
-                        from project.zipTree("${t.destinationDir}/${t.archiveName}")
-                    }
-                }
-            }
-        }
-    }
-
-    @PackageScope
     void addJrubyExtensionToJar(Project project) {
         if(!Jar.metaClass.respondsTo(Jar.class,'jruby',Closure)) {
             Jar.metaClass.jruby = { Closure extraConfig ->
                 JRubyJarConfigurator.configureArchive(delegate,extraConfig)
             }
         }
+        if(!JRubyJar.metaClass.respondsTo(JRubyJar.class,'jruby',Closure)) {
```

Comment:
OK. I think this can be done simpler, becasue we now have control over the task type in thie case of `JRubyJar`, which we do not have with `Jar`.